ruby/lib/ci/queue/build_record.rb

```diff
@@ -0,0 +1,44 @@
+module CI
+  module Queue
+    class BuildRecord
+      attr_reader :error_reports
+
+      def initialize(queue)
+        @queue = queue
+        @error_reports = {}
+        @stats = {}
+      end
+
+      def progress
+        @queue.progress
+      end
+
+      def queue_exhausted?
+        @queue.exhausted?
+      end
+
+      def record_error(id, payload, stats: nil)
+        error_reports[id] = payload
+        record_stats(stats)
+      end
+
+      def record_success(id, stats: nil)
+        error_reports.delete(id)
+        record_stats(stats)
+      end
+
+      def fetch_stats(stat_names)
+        stat_names.zip(stats.values_at(*stat_names).map(&:to_f))
+      end
+
+      private
+
+      attr_reader :stats
```

Comment:
Nitpick: could this be merged with the `attr_reader` at the top of the class?

Reply:
No because I want ti to be private.